### ISSUE TITLE
Problem when reader is disconnected and reconnected

### ISSUE THREAD

**Issue by Fabsky** [perevoznyk/swelio-sdk#1]
Hell0,

First, thank you for your project and for putting this on github !
Then, I come back here concerning my comment and your reply, available here : https://perevoznyk.wordpress.com/2016/05/03/swell-1-9/#comment-337

In resume: when I remove the reader, then add it again, events are not triggered anymore and I've no way to know I've to reload the list of reader to enable events again.

I've tested to set the TraceEvents to true, but it still doesn't work. so here are some pieces of my code:

```
private void startEngine()
        {
            this.Stop();
            this._hasInsertedCard = false;
            this._engine = new Manager();
            this._engine.Active = true;
            this._engine.ReloadReadersList();
            this._engine.CardInserted += engine_CardInserted;
            this._engine.CardRemoved += engine_CardRemoved;
            this._engine.ReadersListChanged += engine_ReadersListChanged;
            this._engine.TraceEvents = true;
        }

        void engine_CardRemoved(object sender, CardEventArgs e)
        {
            try
            {
                this.onCardRemoved();
            }
            catch (Exception exc)
            {
                this.Logger.Error("An error occured when handling the eId card removal", exc);
            }
            finally
            {
                this._engine.TraceEvents = true;
            }
        }

        void engine_CardInserted(object sender, CardEventArgs e)
        {
            try
            {
                this.onCardInserted(e.ReaderNumber);
            }
            catch (Exception exc)
            {
                this.Logger.Error("An error occured when handling the eId card insertion", exc);
            }
            finally
            {
                this._engine.TraceEvents = true;
            }
        }

        void engine_ReadersListChanged(object sender, EventArgs e)
        {
            this._engine.TraceEvents = true;
        }
```

My scenario:
1. Connect the reader
2. Insert a card => method engine_CardInserted called
3. Remove the card => method engine_CardRemoved called
4. Remove the reader

Repeat scenario: this time no method are called.

Now that you've published the code on github, I may try to find some time to look it and fix this, but in the meantime if you've an idea ....

Thank you
Fabrice

**Comment by perevoznyk** [comment]
Hi Fabrice,

Thank you for your detailed report. I will do the test in the next few days and will try to resolve this problem. Also, the current event handling approach is not very clear. I will try to change it. 

Serhiy

**Comment by perevoznyk** [comment]
The problem with removal of card reader in Windows 10 when the single card reader was attached is solved in 1.10.0.0 release.

**Comment by Fabsky** [comment]
Hello @perevoznyk 
Sorry I only use it now, but still have the problem. Maybe back due to windows update (?).
In fact I only get a `ReadersListChanged `the 1st time I connect the reader. Dont get it when I disconnect, nor when I reconnect the reader. Do you still maintain this project ?

The `Swelio.Engine.Manager.EventTracer` is not called at all when I disconnect the USB reader.
